### PR TITLE
Register Map Implementation for Cortex M33

### PR DESCRIPTION
**Pull request by Dolphindalt**
Provides a minimal implementation of `core/src/cpus/cortex_m/m33.zig` derived from the [official documentation](https://www.st.com/resource/en/programming_manual/pm0264-stm32-cortexm33-mcus-programming-manual-stmicroelectronics.pdf).

### CODE REVIEW

**Comment by ikskuh** [comment]
Can you add a link to the documentation into the file? That would help maintainers quickly find the docs if they need to touch that code :)

File: core/src/cpus/cortex_m/m33.zig [hunk 1]
```diff
@@ -1,3 +1,159 @@
...
+    RLAR_A2: RLAR,
+    /// MPU Region Base Address Register Alias 3
+    RLAR_A3: RLAR,
+    _reserved0: [20]u8,
+    /// MPU Memory Addribute Indirection Register 0
+    MPU_MAIR0: u32,
+    /// MPU Memory Addribute Indirection Register 1
+    MPU_MAIR1: u32,
+
+    pub const RBAR = mmio.Mmio(packed struct(u32) {
```
Comment (mattnite) [inline core/src/cpus/cortex_m/m33.zig:146]: Please add doc comments to fields

File: core/src/cpus/cortex_m/m33.zig [hunk 2]
```diff
@@ -1,3 +1,159 @@
...
+        XN: u1,
+        AP: u2,
+        SH: u2,
+        BASE: u27,
+    });
+
+    pub const RLAR = mmio.Mmio(packed struct(u32) {
```
Comment (mattnite) [inline core/src/cpus/cortex_m/m33.zig:153]: Please add doc comments to fields

**Review by mattnite — CHANGES_REQUESTED** [review]

**Review by mattnite — APPROVED** [review]
Thank you!

**Pull request merged**